### PR TITLE
Storage May GA Release

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
Did we bump up any API version (`x-ms-version`) in the beta releases last month?

**Comment by microzchang** [comment]
> Did we bump up any API version (`x-ms-version`) in the beta releases last month?

File Share version from `2021-12-02` to `2022-11-02`

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 ## 12.5.0 (2023-05-09)
 
+### Features Added
+
 - New features in `12.5.0-beta.1` are now generally available.
 
+### Bugs Fixed
+
+- Fixed a bug where `ShareFileClient::ListHandles` and `ShareDirectoryClient::ListHandles` returned an empty list regardless of there are handles or not.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:11]: - Fixed a bug where `ShareFileClient::ListHandles` and `ShareDirectoryClient::ListHandles` always return empty list.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**